File: 2022/day06/go/main.go
package main

import "os"

func main() {
	content, _ := os.ReadFile("puzzle.input")
	sz, signal_size := len(content), 14

	for i := 0; i < sz; i++ {
		end := i + signal_size
		if end >= sz {
			break
		}
		signal := map[string]bool{}
		for _, c := range content[i:end] {
			signal[string(c)] = true
		}
		if len(signal) == signal_size {
			print(end)
			break
		}
	}
}

File: 2022/day06/rust/src/main.rs
use std::collections::HashSet;

fn main() {
    let input = std::fs::read_to_string("puzzle.input")
        .expect("should have been ablet to read puzzle.input.sample");
    let sz = input.len();

    let signal_size = 14;

    for i in 0..sz {
        let end = i + signal_size;
        if end >= sz {
            break;
        }
        let signal: HashSet<char> = input[i..end].chars().collect();
        if signal.len() == signal_size {
            println!("{end}");
            break;
        }
    }
}
